libcontainer/factory_linux.go

```diff
@@ -239,19 +239,23 @@ func (l *LinuxFactory) StartInitialization() (err error) {
 		}
 		*pair.v = i
 	}
+
 	var (
 		pipe = os.NewFile(uintptr(pipefd), "pipe")
 		it   = initType(os.Getenv("_LIBCONTAINER_INITTYPE"))
 	)
+
+	// ensure that this pipe is always closed in the case of failure.
+	defer pipe.Close()
```

Comment:
If you want to make this a `defer` please put it inside the `defer func()` where the old `pipe.Close` used to be. I can't remember the exact reasoning behind it, but you need to make sure that you close `pipe.Close` immediately after sending the error.